pkg/cluster/clusterserviceprincipal.go

```diff
@@ -173,50 +177,50 @@ func (m *manager) updateAROSecret(ctx context.Context) error {
 }
 
 func (m *manager) updateOpenShiftSecret(ctx context.Context) error {
-	var changed bool
-	spp := m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile
-	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		//data:
-		// azure_client_id: secret_id
-		// azure_client_secret: secret_value
-		// azure_tenant_id: tenant_id
-		secret, err := m.kubernetescli.CoreV1().Secrets("kube-system").Get(ctx, "azure-credentials", metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		if string(secret.Data["azure_client_id"]) != spp.ClientID {
-			secret.Data["azure_client_id"] = []byte(spp.ClientID)
-			changed = true
-		}
-
-		if string(secret.Data["azure_client_secret"]) != string(spp.ClientSecret) {
-			secret.Data["azure_client_secret"] = []byte(spp.ClientSecret)
-			changed = true
-		}
-
-		if string(secret.Data["azure_tenant_id"]) != m.subscriptionDoc.Subscription.Properties.TenantID {
-			secret.Data["azure_tenant_id"] = []byte(m.subscriptionDoc.Subscription.Properties.TenantID)
-			changed = true
-		}
-
-		if changed {
-			_, err = m.kubernetescli.CoreV1().Secrets("kube-system").Update(ctx, secret, metav1.UpdateOptions{})
-			if err != nil {
-				return err
-			}
-		}
-		return nil
-	})
-	if err != nil {
+	//data:
+	// azure_client_id: secret_id
+	// azure_client_secret: secret_value
+	// azure_tenant_id: tenant_id
+	secret, err := m.kubernetescli.CoreV1().Secrets(clusterauthorizer.AzureCredentialSecretNameSpace).Get(ctx, clusterauthorizer.AzureCredentialSecretName, metav1.GetOptions{})
```

Comment:
The only reason to `Get` the secret with server-side apply is because of the pod restarts required when the service principal changes. We could also further simplify the logic here by not even caring about whether the credentials are changing. We would define the data we want on the secret and just apply it and restart the `cloud-credential-operator` pods every time this function is run. This code only runs when the cluster admin runs `az aro update`, so restarting the pods shouldn't make much of a difference.